src/app/content/components/Page/highlightManager.ts

```diff
@@ -87,7 +88,7 @@ const highlightData = (services: Services) => (data: HighlightData) => {
 
   const serialized = SerializedHighlight.fromApiResponse(data);
 
-  highlighter.highlight(serialized);
+  attachHighlight(serialized, highlighter);
 
   return highlighter.getHighlight(data.id);
```

Comment:
you could have `attachHighlight` return the highlight object so you don't have to fetch it a second time here